set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java

```diff
@@ -61,13 +61,32 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
 	
 	addTitleLangCriterion(searchCriteria, searchQuery);
 	
-	searchQuery.setSortCriteria(toArray(sort));	
+	addSortCriterion(searchCriteria, searchQuery, sort);
+	
 	searchQuery.setPageSize(pageSize);
 	searchQuery.setPageNr(page);
 	
 	return searchQuery;
     }
 
+    private void addSortCriterion(Map<String, String> searchCriteria, UserSetQuery searchQuery, String sort) throws ParamValidationException {
+      //validate sorting based on score (can be only in the descending order)
+      if(sort!=null && sort.contains(WebUserSetFields.TEXT_SCORE_SORT) && !searchCriteria.containsKey(WebUserSetFields.TEXT)) {
```

Comment:
the parameter validation including the value provided int the sort is the first step to be performed after the authorization. Please move the validation for the score parameter to the existing method for param validation in the controller method